k4Interface/include/k4Interface/ITowerToolThetaModule.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2014-2023 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef RECINTERFACE_ITOWERTOOLTHETAMODULE_H
+#define RECINTERFACE_ITOWERTOOLTHETAMODULE_H
+
+// Gaudi
+#include "GaudiKernel/IAlgTool.h"
+
+// datamodel
+#include "edm4hep/CalorimeterHitCollection.h"
+#include "edm4hep/Cluster.h"
+
+/** @class ITowerToolThetaModule RecInterface/RecInterface/ITowerToolThetaModule.h ITowerToolThetaModule.h
+ *
+ *  Abstract interface to tower building tool.
+ *
+ *  @author Anna Zaborowska
+ *  @Modified by Tong Li (t.li at cern.ch), for sliding-window clustering at FCCee where the CaloTower is created from Theta-Phi
+ */
+
+class ITowerToolThetaModule : virtual public IAlgTool {
```

Comment:
Probably it should have a virtual destructor

Reply:
@jmcarcell could you provide a simple example?